lib/oauth2/client.rb

```diff
@@ -25,23 +25,27 @@ class Client # rubocop:disable Metrics/ClassLength
     # @option opts [Symbol] :auth_scheme (:basic_auth) HTTP method to use to authorize request (:basic_auth or :request_body)
     # @option opts [Hash] :connection_opts ({}) Hash of connection options to pass to initialize Faraday with
     # @option opts [FixNum] :max_redirects (5) maximum number of redirects to follow
-    # @option opts [Boolean] :raise_errors (true) whether or not to raise an OAuth2::Error
-    #  on responses with 400+ status codes
+    # @option opts [Boolean] :raise_errors (true) whether or not to raise an OAuth2::Error on responses with 400+ status codes
+    # @option opts [Proc] :extract_access_token  proc that extracts the access token from the response
```

Comment:
Please add more comment to remind this proc returning an AccessToken/AccessToken-like object.

Reply:
Is there a known pattern for Yard/RDoc to document "Callable" params/options like this? I think we want to declare the signature and the return value. Can just do it in text but I don't know if there is a better way.